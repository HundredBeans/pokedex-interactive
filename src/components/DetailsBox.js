import React from "react";
import {
  Flex,
  HStack,
  Tag,
  TagLabel,
  Container,
  Image,
  SimpleGrid,
  Heading,
  Stat,
  StatLabel,
  StatNumber,
  Divider,
  GridItem,
  Text,
} from "@chakra-ui/react";
import placeholder from "../assets/placeholder.png";
import ErrorAlert from "./ErrorAlert";

const CommonInfo = (props) => {
  return (
    <Flex flexDirection="column" align="center">
      <HStack p={3}>
        {props.types.map((item, index) => (
          <Tag key={index} variant="subtle" size="lg" p={2} colorScheme="green">
            <TagLabel fontSize={18} textTransform="capitalize">
              {item.type.name}
            </TagLabel>
          </Tag>
        ))}
      </HStack>
      <Container
        borderColor="teal"
        display="flex"
        justifyContent="space-around"
        alignItems="center"
        flexDirection={{ base: "column", sm: "row" }}
      >
        <Image fallbackSrc={placeholder} boxSize="250px" src={props.imageSrc} />
        <Divider orientation="vertical" />

        <SimpleGrid
          columns={3}
          spacing={5}
          alignItems="center"
          textAlign="center"
        >
          <GridItem colSpan={3}>
            <Heading as="h4">
              {props.pokemonName.toString().toUpperCase()}
            </Heading>
          </GridItem>
          <Stat>
            <StatLabel>ID</StatLabel>
            <StatNumber># {props.id}</StatNumber>
          </Stat>

          <Stat>
            <StatLabel>Height</StatLabel>
            <StatNumber>{props.height}</StatNumber>
          </Stat>

          <Stat>
            <StatLabel>Weight</StatLabel>
            <StatNumber>{props.weight}</StatNumber>
          </Stat>
          {props.caughtDate && (
            <GridItem colSpan={3}>
              <Text color="blue">Caught at {props.caughtDate}</Text>
            </GridItem>
          )}
        </SimpleGrid>
      </Container>
    </Flex>
  );
};

const MovesList = (props) => {
  return (
    <Container display="flex" flexDirection="column" textAlign="center">
      <Heading p={5}>Moves List</Heading>

      <SimpleGrid
        marginInline={3}
        columns={3}
        spacing={5}
        alignItems="center"
        maxHeight="200px"
        overflowY="scroll"
        className="no-scrollbar"
      >
        {props.moves.length === 0 ? (
          <GridItem colSpan={3}>
            <ErrorAlert errorTitle="Data Empty" />
          </GridItem>
        ) : (
          props.moves.map((item, index) => (
            <Tag
              key={index}
              variant="subtle"
              size="sm"
              p={2}
              colorScheme="blue"
            >
              <TagLabel fontSize={18} textTransform="capitalize">
                {item.move.name}
              </TagLabel>
            </Tag>
          ))
        )}
      </SimpleGrid>
    </Container>
  );
};

function DetailsBox(props) {
  return (
    <React.Fragment>
      <CommonInfo {...props} />
      <MovesList {...props} />
    </React.Fragment>
  );
}

export default DetailsBox;
