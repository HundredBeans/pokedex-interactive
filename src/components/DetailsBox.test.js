import { cleanup } from "@testing-library/react";
import React from "react";
import { render } from "../test-utils";
import DetailsBox from "./DetailsBox";

describe("DetailsBox.js test", () => {
  afterEach(() => {
    cleanup();
  });

  it("renders the component (pokemonDetails without caughtDate)", () => {
    const pokemonData = {
      height: 17,
      id: 87,
      moves: [
        {
          move: {
            url: "https://pokeapi.co/api/v2/move/6/",
            name: "pay-day",
            __typename: "BaseName",
          },
          __typename: "Move",
        },
      ],
      name: "dewgong",
      sprites: {
        front_default:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/87.png",
        __typename: "Sprite",
      },
      types: [
        {
          slot: 1,
          type: {
            url: "https://pokeapi.co/api/v2/type/11/",
            name: "water",
            __typename: "BaseName",
          },
          __typename: "Type",
        },
        {
          slot: 2,
          type: {
            url: "https://pokeapi.co/api/v2/type/15/",
            name: "ice",
            __typename: "BaseName",
          },
          __typename: "Type",
        },
      ],
      weight: 1200,
      message: "",
      timeCaught: "3/28/2021, 7:05:48 PM",
      __typename: "Pokemon",
    };
    const renderedComponent = render(
      <DetailsBox
        imageSrc={pokemonData.sprites.front_default}
        id={pokemonData.id}
        types={pokemonData.types}
        height={pokemonData.height}
        weight={pokemonData.weight}
        moves={pokemonData.moves}
        pokemonName={pokemonData.name}
      />
    );
    expect(renderedComponent).toMatchSnapshot("without-caughtDate");
    renderedComponent.rerender(
      <DetailsBox
        imageSrc={pokemonData.sprites.front_default}
        id={pokemonData.id}
        types={pokemonData.types}
        height={pokemonData.height}
        weight={pokemonData.weight}
        moves={pokemonData.moves}
        pokemonName={pokemonData.name}
        caughtDate={pokemonData.timeCaught}
      />
    );
    expect(renderedComponent).toMatchSnapshot("with-caughtDate");
  });
});
