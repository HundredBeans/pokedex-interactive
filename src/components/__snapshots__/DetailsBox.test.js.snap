// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsBox.js test renders the component (pokemonDetails without caughtDate): with-caughtDate 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-dvxtzn"
      >
        <div
          class="chakra-stack css-kuxzr6"
        >
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              water
            </span>
          </span>
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              ice
            </span>
          </span>
        </div>
        <div
          class="chakra-container css-ju1cjz"
        >
          <img
            class="chakra-image__placeholder css-1tz2n1v"
            src="placeholder.png"
          />
          <hr
            aria-orientation="vertical"
            class="chakra-divider css-1aam7vk"
            role="separator"
          />
          <div
            class="css-10sffrr"
          >
            <div
              class="css-rklm6r"
            >
              <h4
                class="chakra-heading css-1dklj6k"
              >
                DEWGONG
              </h4>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  ID
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  # 
                  87
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Height
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  17
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Weight
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  1200
                </dd>
              </dl>
            </div>
            <div
              class="css-rklm6r"
            >
              <p
                class="chakra-text css-14ksm7b"
              >
                Caught at 
                3/28/2021, 7:05:48 PM
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-container css-3aul1p"
      >
        <h2
          class="chakra-heading css-ci02qy"
        >
          Moves List
        </h2>
        <div
          class="no-scrollbar css-iqyxvu"
        >
          <span
            class="css-1fza2wb"
          >
            <span
              class="css-g4o07i"
            >
              pay-day
            </span>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dvxtzn"
    >
      <div
        class="chakra-stack css-kuxzr6"
      >
        <span
          class="css-mi7bwv"
        >
          <span
            class="css-1v3sf2x"
          >
            water
          </span>
        </span>
        <span
          class="css-mi7bwv"
        >
          <span
            class="css-1v3sf2x"
          >
            ice
          </span>
        </span>
      </div>
      <div
        class="chakra-container css-ju1cjz"
      >
        <img
          class="chakra-image__placeholder css-1tz2n1v"
          src="placeholder.png"
        />
        <hr
          aria-orientation="vertical"
          class="chakra-divider css-1aam7vk"
          role="separator"
        />
        <div
          class="css-10sffrr"
        >
          <div
            class="css-rklm6r"
          >
            <h4
              class="chakra-heading css-1dklj6k"
            >
              DEWGONG
            </h4>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                ID
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                # 
                87
              </dd>
            </dl>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                Height
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                17
              </dd>
            </dl>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                Weight
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                1200
              </dd>
            </dl>
          </div>
          <div
            class="css-rklm6r"
          >
            <p
              class="chakra-text css-14ksm7b"
            >
              Caught at 
              3/28/2021, 7:05:48 PM
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-container css-3aul1p"
    >
      <h2
        class="chakra-heading css-ci02qy"
      >
        Moves List
      </h2>
      <div
        class="no-scrollbar css-iqyxvu"
      >
        <span
          class="css-1fza2wb"
        >
          <span
            class="css-g4o07i"
          >
            pay-day
          </span>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DetailsBox.js test renders the component (pokemonDetails without caughtDate): without-caughtDate 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-dvxtzn"
      >
        <div
          class="chakra-stack css-kuxzr6"
        >
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              water
            </span>
          </span>
          <span
            class="css-mi7bwv"
          >
            <span
              class="css-1v3sf2x"
            >
              ice
            </span>
          </span>
        </div>
        <div
          class="chakra-container css-ju1cjz"
        >
          <img
            class="chakra-image__placeholder css-1tz2n1v"
            src="placeholder.png"
          />
          <hr
            aria-orientation="vertical"
            class="chakra-divider css-1aam7vk"
            role="separator"
          />
          <div
            class="css-10sffrr"
          >
            <div
              class="css-rklm6r"
            >
              <h4
                class="chakra-heading css-1dklj6k"
              >
                DEWGONG
              </h4>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  ID
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  # 
                  87
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Height
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  17
                </dd>
              </dl>
            </div>
            <div
              class="chakra-stat css-1mbo1ls"
            >
              <dl>
                <dt
                  class="chakra-stat__label css-14go5ty"
                >
                  Weight
                </dt>
                <dd
                  class="chakra-stat__number css-1axeus7"
                >
                  1200
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-container css-3aul1p"
      >
        <h2
          class="chakra-heading css-ci02qy"
        >
          Moves List
        </h2>
        <div
          class="no-scrollbar css-iqyxvu"
        >
          <span
            class="css-1fza2wb"
          >
            <span
              class="css-g4o07i"
            >
              pay-day
            </span>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dvxtzn"
    >
      <div
        class="chakra-stack css-kuxzr6"
      >
        <span
          class="css-mi7bwv"
        >
          <span
            class="css-1v3sf2x"
          >
            water
          </span>
        </span>
        <span
          class="css-mi7bwv"
        >
          <span
            class="css-1v3sf2x"
          >
            ice
          </span>
        </span>
      </div>
      <div
        class="chakra-container css-ju1cjz"
      >
        <img
          class="chakra-image__placeholder css-1tz2n1v"
          src="placeholder.png"
        />
        <hr
          aria-orientation="vertical"
          class="chakra-divider css-1aam7vk"
          role="separator"
        />
        <div
          class="css-10sffrr"
        >
          <div
            class="css-rklm6r"
          >
            <h4
              class="chakra-heading css-1dklj6k"
            >
              DEWGONG
            </h4>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                ID
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                # 
                87
              </dd>
            </dl>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                Height
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                17
              </dd>
            </dl>
          </div>
          <div
            class="chakra-stat css-1mbo1ls"
          >
            <dl>
              <dt
                class="chakra-stat__label css-14go5ty"
              >
                Weight
              </dt>
              <dd
                class="chakra-stat__number css-1axeus7"
              >
                1200
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-container css-3aul1p"
    >
      <h2
        class="chakra-heading css-ci02qy"
      >
        Moves List
      </h2>
      <div
        class="no-scrollbar css-iqyxvu"
      >
        <span
          class="css-1fza2wb"
        >
          <span
            class="css-g4o07i"
          >
            pay-day
          </span>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
